Extensions/PlatformBehavior/platformerobjectruntimebehavior.ts

```diff
@@ -478,37 +499,34 @@ namespace gdjs {
 
     private _checkTransitionOnFloorOrFalling() {
       const object = this.owner;
-      //Check if the object is on a floor:
-      //In priority, check if the last floor platform is still the floor.
-      let oldY = object.getY();
-      object.setY(object.getY() + 1);
-      if (
-        this._state === this._onFloor &&
-        gdjs.RuntimeObject.collisionTest(
-          object,
-          this._onFloor.getFloorPlatform()!.owner,
-          this._ignoreTouchingEdges
-        )
-      ) {
-        //Still on the same floor
-        this._onFloor.updateFloorPosition();
-      } else {
-        // Avoid landing on a platform if the object is not going down.
-        // (which could happen for Jumpthru, when the object jump and pass just at the top
-        // of a jumpthru, it could be considered as landing if not this this extra check).
-        const canLand = this._requestedDeltaY >= 0;
-
-        //Check if landing on a new floor: (Exclude already overlapped jump thru)
-        let collidingPlatform = this._getCollidingPlatform();
-        if (canLand && collidingPlatform !== null) {
-          //Register the colliding platform as the floor.
-          this._setOnFloor(collidingPlatform);
-        } else if (this._state === this._onFloor) {
-          // don't fall if GrabbingPlatform or OnLadder
+      const oldY = object.getY();
+      // Avoid landing on a platform if the object is not going down.
+      // (which could happen for Jumpthru, when the object jump and pass just at the top
+      // of a jumpthru, it could be considered as landing if not this this extra check).
+      const canLand = this._requestedDeltaY >= 0;
+
+      // The interval could be smaller.
+      // It's just for rounding errors.
+      const floor = this._findHighestFloorAndMoveOnTop(
+        this._potentialCollidingObjects,
+        -1,
+        1
```

Comment:
Do you think that for rounding errors we should add an epsilon too each (-1-epsilon, 1+epsilon)?

Reply:
I'm asking because I found a owner that was with these hitboxes: 
![image](https://user-images.githubusercontent.com/1280130/133696447-4d98b04d-c50c-44a0-aa73-beae8270c599.png)

Giving this `floorMaxY`:
![image](https://user-images.githubusercontent.com/1280130/133696498-03ff922d-c125-415b-af0c-729781cdeb40.png)

And unfortunately the floor is: 
![image](https://user-images.githubusercontent.com/1280130/133696567-11288125-1fac-42d2-8af4-9f7ec3d3884e.png)

And his discarded because of the 1191.36 being above 1191.359999997

This was causing problem when the object changed height. I assume that when the object change height, the rounding errors is appearing when the object is repositioned.

Reply:
I prefer a bigger margin because of the separate call that could move the object in unexpected ways. I have a bad feeling about this call because it can move the owner in any direction so it's hard to predict what it will do. I would like to try to replace this call and the remaining pixel loops one day.

Reply:
>  So, the 1 pixel margin is already a big margin. This makes me thing what is on the other comment should be the fix.

You're correct, I've removed the `- 1` as discussed on the other comment and don't have this issue anymore.
So this is not a problem anymore, the rounding issues were coming from the height changes repositioning that was putting the object too far (1 pixel away, which was exactly the size of this margin).

> the separate call that could move the object in unexpected ways

What do you mean exactly by "separate call" here?

Reply:
```JavaScript
      //Ensure the object is not stuck
      if (this._separateFromPlatforms(this._potentialCollidingObjects, true)) {
        //After being unstuck, the object must be able to jump again.
        this._canJump = true;
      }
```
I'm not sure if it can be made to work without it though.

Reply:
Not sure indeed we can do without. The "invariant" for a platformer object is that it is never colliding with a platform - if that's the case we assume we got stuck? At least that was the original idea. Not sure if this still holds if you consider platforms moving on the Y axis, but seems that it's working well despite this?
Though maybe this would be something to move later in the future, so that we've already moved the object on all axes before trying to unstick the item if it's inside a platform...